File: test/func.c
#include<stdbool.h>
#include<stdio.h>

const int a = 5; 
const int b = -6; 
const char c = 't'; 

int x, y, d; 
char e[50]; 
struct {
	float f1; 
	bool f2[50][100]; 
	struct {
		float f31; 
		bool f32[50][100]; 
		struct {
			float f331; 
			bool f332[50][100]; 
			int f333; 
		}
		f33; 
	}
	f3; 
}
f; 
bool g[50][100]; 

int func(){
	int i; 
	
	if(b == 0){
		return a; 
	}
	else{
		return func(a, b); 
	}
	for(i = 1; i < a; i++){
		printf("b: %d\n", b); 
	}
	i = 0; 
	while(i < 5){
		printf("b: %d\n", b); 
		i = i + 1; 
	}
	func(a, b); 
	func(i, a); 
	func(i, a, b, c, d); 
}

int main(int argc, char* argv[]){
	scanf("%d%d", &x, &y); 
	printf("x: %d\ny: %d\n", x, y); 
	func(x, y); 
	func(x); 
}

File: test/func.pas
Program example(input, output);
Const 
    a = 5; 
    b = -6; 
    c = 't';
Var 
    x, y, d : integer;
    e : array [1..50] of char;
    f : record
        f1 : real;
        f2 : array [1..50, 1..100] of boolean;
        f3 : record
            f31 : real;
            f32 : array [1..50, 1..100] of boolean;
            f33 : record
                f331 : real;
                f332 : array [1..50, 1..100] of boolean;
                f333 : integer;
            End;
        End;
    End;
    g : array [1..50, 1..100] of boolean;
Function func : integer;
var i:integer;
Begin
    If b = 0 Then func := a
    Else func := func(a,b);
    for i:=1 to a do
        begin
            write(b);
        end;
    i:=0;
    While i<5 Do
        begin
            write(b);
            i:=i+1;
        end;
    func(a,b);
    func(i,a);
    func(i,a,b,c,d)
End;
Begin
    read(x,y);
    write(x,y);
    func(x, y);
    func(x)
End.